docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,692 @@
+
+### <xref:System.Net.ServicePoint> Properties Mapping
+
+| <xref:System.Net.ServicePoint> Old API | New API | Notes |
+|---------|----------------------|-------|
+| `Address` | `HttpRequestMessage.RequestUri` | This is request uri, this information can be found under `HttpRequestMessage`. |
+| `BindIPEndPointDelegate` | No direct equivalent API | See <xref:System.Net.Http.SocketsHttpHandler.ConnectCallback>. |
+| `Certificate` | No direct equivalent API | This information can be fetched from `RemoteCertificateValidationCallback`. See: [Example: Fetch Certificate](#example-fetch-certificate) |
+| `ClientCertificate` | No equivalent API | TODO |
+| `ConnectionLeaseTimeout` | `SocketsHttpHandler.PooledConnectionLifetime` | Equivalent setting in <xref:System.Net.Http.HttpClient> |
+| `ConnectionLimit` | <xref:System.Net.Http.SocketsHttpHandler.MaxConnectionsPerServer> | TODO |
+| `ConnectionName` | No equivalent API | TODO |
+| `CurrentConnections` | No equivalent API | TODO |
+| `Expect100Continue` | <xref:System.Net.Http.Headers.HttpRequestHeaders.ExpectContinue> | TODO |
+| `IdleSince` | No equivalent API | TODO |
+| `MaxIdleTime` | <xref:System.Net.Http.SocketsHttpHandler.PooledConnectionIdleTimeout> | TODO |
+| `ProtocolVersion` | `HttpRequestMessage.Version` | We can get this from `HttpRequestMessage`. |
+| `ReceiveBufferSize` | No direct equivalent API | See <xref:System.Net.Http.SocketsHttpHandler.ConnectCallback>. |
+| `SupportsPipelining` | No equivalent API | `HttpClient` doesn't support pipelining. |
+| `UseNagleAlgorithm` | No direct equivalent API | See <xref:System.Net.Http.SocketsHttpHandler.ConnectCallback>. |
+
+### <xref:System.Net.ServicePoint> Method Mapping
+
+| <xref:System.Net.ServicePoint> Old API | New API | Notes |
+|---------|----------------------|-------|
+| `CloseConnectionGroup` | No equivalent | No workaround |
+| `SetTcpKeepAlive` | No direct equivalent API | See <xref:System.Net.Http.SocketsHttpHandler.ConnectCallback>. |
+
+## Usage of HttpClient and HttpRequestMessage Properties
+
+When working with HttpClient in .NET, you have access to a variety of properties that allow you to configure and customize HTTP requests and responses. Understanding these properties can help you make the most of HttpClient and ensure that your application communicates efficiently and securely with web services.
+
+### Example: Usage of HttpRequestMessage properties
+
+Here's an example of how to use HttpClient and HttpRequestMessage together:
+
+```csharp
+var client = new HttpClient();
+
+var request = new HttpRequestMessage(HttpMethod.Post, "https://example.com"); // Method and RequestUri usage
+var request = new HttpRequestMessage() // Alternative way to set RequestUri and Method
+{
+    RequestUri = new Uri("https://example.com"),
+    Method = HttpMethod.Post
+};
+request.Headers.Add("Custom-Header", "value");
+request.Content = new StringContent("somestring");
+
+var response = await client.SendAsync(request);
```

Comment:
```suggestion
using var response = await client.SendAsync(request);
```

Similar concern as https://github.com/dotnet/docs/pull/42242#discussion_r1753860667, we should try to dispose responses in all example code since people will just copy-paste them